code/modules/food_and_drinks/kitchen_machinery/griddle.dm

```diff
@@ -61,19 +66,26 @@
 	item_to_grill.flags_1 |= IS_ONTOP_1
 	RegisterSignal(item_to_grill, COMSIG_MOVABLE_MOVED, .proc/ItemMoved)
 	RegisterSignal(item_to_grill, COMSIG_GRILL_COMPLETED, .proc/GrillCompleted)
+	RegisterSignal(item_to_grill, COMSIG_PARENT_QDELETING, .proc/ItemDeleted)
 	update_grill_audio()
 
-/obj/machinery/griddle/proc/ItemMoved(obj/item/I, atom/OldLoc, Dir, Forced)
-	SIGNAL_HANDLER
+/obj/machinery/griddle/proc/ItemRemovedFromGrill(obj/item/I)
 	I.flags_1 &= ~IS_ONTOP_1
 	griddled_objects -= I
 	vis_contents -= I
-	UnregisterSignal(I, COMSIG_GRILL_COMPLETED)
+	UnregisterSignal(I, list(COMSIG_GRILL_COMPLETED, COMSIG_MOVABLE_MOVED, COMSIG_PARENT_QDELETING))
 	update_grill_audio()
 
+/obj/machinery/griddle/proc/ItemMoved(obj/item/I, atom/OldLoc, Dir, Forced)
+	SIGNAL_HANDLER
+	ItemRemovedFromGrill(I)
+
+/obj/machinery/griddle/proc/ItemDeleted(datum/source)
+	SIGNAL_HANDLER
+	ItemRemovedFromGrill(source)
```

Comment:
Why not `RegisterSignal(item_to_grill, COMSIG_PARENT_QDELETING, .proc/ItemRemovedFromGrill)`?